presto-main/src/main/java/com/facebook/presto/connector/informationSchema/InformationSchemaMetadata.java

```diff
@@ -190,11 +212,103 @@ public ColumnMetadata getColumnMetadata(ConnectorSession session, ConnectorTable
+            ConnectorSession connectorSession,
+            TupleDomain<ColumnHandle> constraint,
+            Predicate<Map<ColumnHandle, NullableValue>> predicate)
+    {
+        Optional<Set<String>> schemas = stringFilter(constraint, SCHEMA_COLUMN_HANDLE);
+
+        if (schemas.isPresent()) {
+            return schemas.get().stream()
+                    .map(schema -> new QualifiedTablePrefix(catalogName, schema))
+                    .collect(toImmutableSet());
+        }
+
+        Session session = ((FullConnectorSession) connectorSession).getSession();
+        return metadata.listSchemaNames(session, catalogName).stream()
+                .filter(schema -> predicate.test(ImmutableMap.of(
+                        CATALOG_COLUMN_HANDLE, new NullableValue(createUnboundedVarcharType(), utf8Slice(catalogName)),
+                        SCHEMA_COLUMN_HANDLE, new NullableValue(createUnboundedVarcharType(), utf8Slice(schema)))))
+                .map(schema -> new QualifiedTablePrefix(catalogName, schema))
+                .collect(toImmutableSet());
+    }
+
+    public Set<QualifiedTablePrefix> calculatePrefixesWithTableName(
+            ConnectorSession connectorSession,
+            Set<QualifiedTablePrefix> prefixes,
+            TupleDomain<ColumnHandle> constraint,
+            Predicate<Map<ColumnHandle, NullableValue>> predicate)
+    {
+        Optional<Set<String>> tables = stringFilter(constraint, TABLE_COLUMN_HANDLE);
+        ImmutableSet.Builder<QualifiedTablePrefix> result = ImmutableSet.builder();
+        if (tables.isPresent()) {
+            for (QualifiedTablePrefix prefix : prefixes) {
+                tables.get().stream()
+                        .map(table -> new QualifiedTablePrefix(prefix.getCatalogName(), prefix.getSchemaName().get(), table))
+                        .forEach(result::add);
+            }
+        }
+        else {
+            Session session = ((FullConnectorSession) connectorSession).getSession();
+            for (QualifiedTablePrefix prefix : prefixes) {
+                Stream.concat(
+                        metadata.listTables(session, prefix).stream(),
+                        metadata.listViews(session, prefix).stream())
+                        .filter(objectName -> predicate.test(ImmutableMap.of(
+                                CATALOG_COLUMN_HANDLE, new NullableValue(createUnboundedVarcharType(), utf8Slice(catalogName)),
+                                SCHEMA_COLUMN_HANDLE, new NullableValue(createUnboundedVarcharType(), utf8Slice(objectName.getSchemaName())),
+                                TABLE_COLUMN_HANDLE, new NullableValue(createUnboundedVarcharType(), utf8Slice(objectName.getObjectName())))))
+                        .map(QualifiedObjectName::asQualifiedTablePrefix)
+                        .forEach(result::add);
+            }
+        }
+        return result.build();
+    }
+
+    public static <T> Optional<Set<String>> stringFilter(TupleDomain<T> constraint, T column)
```

Comment:
this won't be used in `LIKE` predicate, right?

Reply:
I removed that method